pkg/istio-agent/agent.go

```diff
@@ -406,13 +406,30 @@ func (a *Agent) Run(ctx context.Context) (func(), error) {
 		return nil, fmt.Errorf("failed to start local DNS server: %v", err)
 	}
 
-	a.secretCache, err = a.newSecretManager()
-	if err != nil {
-		return nil, fmt.Errorf("failed to start workload secret manager %v", err)
-	}
+	if socketFileExists(security.WorkloadIdentitySocketPath) {
+		log.Info("SDS socket found. Istio SDS Server won't be started")
+	} else {
+		if security.CheckWorkloadCertificate(security.WorkloadIdentityCertChainPath, security.WorkloadIdentityKeyPath, security.WorkloadIdentityRootCertPath) {
```

Comment:
Not sure who is responsible to provide certs in env rather than gke

Reply:
SPIRE is adding it, and GKE already does. Any CSI driver or similar could implement the same as long as they get the certs into the well known file path

Reply:
Oh, that means spire can provide certs to the dir or serve them through  uds

Reply:
SPIRE using a helper and other CAs could mount the certs on the dir. The main and recommended integration with SPIRE is through UDS.

Reply:
got it thanks